Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -228,6 +240,51 @@ struct Gx2FitterResult {
   const TrackingVolume* startVolume = nullptr;
 };
 
+/// @brief A container to store scattering properties for each material surface
+///
+/// This struct holds the scattering angles, the inverse covariance of the
+/// material, and a validity flag indicating whether the material is valid for
+/// the scattering process.
+struct ScatteringProperties {
+ public:
+  /// @brief Constructor to initialize scattering properties.
+  ///
+  /// @param scatteringAngles_ The vector of scattering angles.
+  /// @param invCovarianceMaterial_ The inverse covariance of the material.
+  /// @param materialIsValid_ A boolean flag indicating whether the material is valid.
+  ScatteringProperties(const BoundVector& scatteringAngles_,
+                       const ActsScalar invCovarianceMaterial_,
+                       const bool materialIsValid_)
+      : scatteringAngles(scatteringAngles_),
+        invCovarianceMaterial(invCovarianceMaterial_),
+        materialIsValid(materialIsValid_) {}
+
+  // Accessor for the scattering angles.
+  const BoundVector& getScatteringAngles() const { return scatteringAngles; }
+
+  // Accessor for a modifiable reference to the scattering angles
+  BoundVector& modifyScatteringAngles() { return scatteringAngles; }
+
+  // Accessor for the inverse covariance of the material.
+  ActsScalar getInvCovarianceMaterial() const { return invCovarianceMaterial; }
```

Comment:
nitpick: usually we call them after the property without `get` and also directly overload with non const so the compiler can choose